sdk/core/src/main/java/com/klaviyo/coresdk/push/KlaviyoPushService.kt

```diff
@@ -0,0 +1,18 @@
+package com.klaviyo.coresdk.push
+
+import com.google.firebase.messaging.FirebaseMessagingService
+import com.klaviyo.coresdk.utils.ConfigFileUtils
+import com.klaviyo.coresdk.utils.ConfigKeys
+
+class KlaviyoPushService: FirebaseMessagingService() {
+    companion object {
+        fun getPushToken(): String {
+            return ConfigFileUtils.readValue(ConfigKeys.PUSH_TOKEN_KEY)
+        }
+    }
+    override fun onNewToken(newToken: String) {
+        super.onNewToken(newToken)
+
+        ConfigFileUtils.writeConfigValue(ConfigKeys.PUSH_TOKEN_KEY, newToken)
```

Comment:
We store push tokens to file. This file is stored in the app's directory so it is cleared out whenever app data is, allowing us. This is in line with how Firebase push tokens are generated and stored too

Reply:
actually don't we want to enqueue a request here?

Reply:
Also are we making this public now? I don't want developers to think they should access this class direclty.

Reply:
The config stuff is still internal but I guess you are worried about if they are overriding this service and seeing this logic in place that it's misleading? We should make sure to store the token somewhere though, or else make an async call to fetch it each time the app is opened and store it in memory.

Reply:
I'll enqueue an identify call here though ya. You are right, didn't realise the iOS SDK was doing that straight away whenever a new token is set.